### ISSUE TITLE
Destructuring default value should be triggered by undefined and not has property

### ISSUE THREAD

**Issue by arv** [google/traceur-compiler#1013]
``` js
var x;
({x = 1} = {});
```

Is currently transformed to:

``` js
var x;
(($__8 = {}, x = "x" in $__8 ? $__8.x : 1, $__8));
```

The semantics has changed to compare the value with undefined instead of doing an in check (I cannot find when it changed):

``` js
var x;
(($0 = {}, $1 = $0.x,  $1 === undefined ? 1 : $1));
```

See #1012
